app/src/main/java/pl/droidevs/books/model/Book.java

```diff
@@ -1,36 +1,67 @@
 package pl.droidevs.books.model;
 
-public class Book {
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
 
-    private String title;
-    private String author;
+public final class Book {
+    private final BookId id;
+    private final String title;
+    private final String author;
+    private final Category category;
     private String description;
+    private String imageUrl;
 
-    public Book(String title) {
+    public Book(@NonNull String title, @NonNull String author, @NonNull Category category) {
+        this(null, title, author, category);
+    }
+
+    public Book(@Nullable BookId id, @NonNull String title, @NonNull String author, @NonNull Category category) {
+        this.id = id;
         this.title = title;
+        this.author = author;
+        this.category = category;
     }
 
-    public String getTitle() {
-        return title;
+    @Nullable
+    public BookId getId() {
+        return id;
     }
 
-    public void setTitle(String title) {
-        this.title = title;
+    @NonNull
+    public String getTitle() {
+        return title;
     }
 
+    @NonNull
     public String getAuthor() {
         return author;
     }
 
-    public void setAuthor(String author) {
-        this.author = author;
+    @NonNull
+    public Category getCategory() {
+        return category;
     }
 
+    @Nullable
     public String getDescription() {
         return description;
     }
 
-    public void setDescription(String description) {
+    public void setDescription(@Nullable String description) {
         this.description = description;
     }
+
+    @Nullable
+    public String getImageUrl() {
+        return imageUrl;
+    }
+
+    public void setImageUrl(@Nullable String imageUrl) {
+        this.imageUrl = imageUrl;
+    }
+
+    public enum Category {
```

Comment:
Uwielbiam **Enumy** - dzięki nim kod zawsze wydawał się taki czytelny. Używałam ich do czasu przeczytania jakiegoś artykułu, w którym wspomnieli, że w androidzie lepiej unikać.

> For example, enums often require more than twice as much memory as static constants. You should strictly avoid using enums on Android.

Cytat z https://developer.android.com/topic/performance/memory.html#Overhead

Zdaję sobie sprawę, że prawdopodobnie nie ma to aż takiego znaczenia w naszym przypadku, ale warto mieć to w głowie oraz rozważyć, czy nie lepiej użyć statycznych intów odpowiednio nazwanych.

Reply:
Wydaje mi się, że korzyści jakie niosą ze sobą enumy przewyższa problemy z większym narzutem pamięciowym. Szczególnie, że obecnie telefony mają po 2-4GB RAM.

Reply:
True! To zawsze jest jakiś kompromis :)